ega-data-api-dataedge/src/main/java/eu/elixir/ega/ebi/dataedge/service/internal/LocalEGAFileLengthServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2016 ELIXIR EGA
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package eu.elixir.ega.ebi.dataedge.service.internal;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import eu.elixir.ega.ebi.dataedge.service.FileLengthService;
+import eu.elixir.ega.ebi.shared.dto.File;
+
+/**
+ * @author amohan
+ */
+@Service
+@Profile("LocalEGA")
+public class LocalEGAFileLengthServiceImpl  implements FileLengthService {
+
+    @Override
+    public long getContentLength(File reqFile, String destinationFormat, long startCoordinate, long endCoordinate) {
+        long length = 0;
+
+        if (startCoordinate > 0 || endCoordinate > 0) {
+            length = endCoordinate - startCoordinate;
+        } else {
+            length = reqFile.getFileSize();

```

Comment:
Currently, when the file is stored in a LocalEga, the sha256sum is added at the beginning  so I think the length would be -32. Also it does not deal with the case of having encryption. I'll also try to test the encrypted part and report back.

Reply:
from my tests was `-32` as well see https://github.com/EGA-archive/ega-data-api/blob/test/m4/ega-data-api-dataedge/src/main/java/eu/elixir/ega/ebi/dataedge/service/internal/RemoteFileServiceImpl.java#L782

I assume @anandmohan777 found some magic way :) to fix it

Reply:
It is working for my ingested file from data-in, I am able to download the whole file with this logic. My understanding is the file stored in s3 is not having any header, it's header value is stored in the database and the value of file size should be the exact value what we need to download. So yes pls check it from your scenario and database value and let me know if it works for -32 I will update it.

Reply:
With
```java
            length = reqFile.getFileSize() - 32;
```
Profile LocalEGA and also

in  private URI getResUri(...)
```java

        } else if (startCoord == 0 && endCoord == 0) {
            builder = UriComponentsBuilder.fromHttpUrl(url)
                    .queryParam("destinationFormat", destFormat)
                    .queryParam("destinationKey", destKey)
                    .queryParam("destinationIV", destIV) <<- new line
                    .queryParam("filePath", fileStableIdPath); // TEST!!
```

to allow to pass the iv to RES I can successfully download.

See https://gist.github.com/omllobet/a0ec2f3206c50b214cfc2f2f4f7b3006

Note also that I noticed that we have two aes "file formats":
* one with iv + aesctrcontent (a.k.a .cip)
* one that just returns aesctrcontent in localega (the iv has to be known by the client to be able to decrypt it)

Reply:
I understand that is working because we are specifying more bytes but then the server closes the connection after sending all the bytes it has and the client for example curl returns a 18

> CURLE_PARTIAL_FILE (18)
> 
> A file transfer was shorter or larger than expected. This happens when the server first reports an expected transfer size, and then delivers data that doesn't match the previously given size.
> 

But the file on disk is ok.

Reply:
Done the changes.